textfsm.py

```diff
@@ -171,13 +171,37 @@ class Key(OptionBase):
     """Value constitutes part of the Key of the record."""
 
   class List(OptionBase):
-    """Value takes the form of a list."""
+    """
+    Value takes the form of a list.
+
+    If the value regex contains mested match groups in the form (?P<name>regex),
+    instead of adding a string to the list, we add a dictionary of the groups.
+
+    Eg.
+    Value List ((?P<name>\w+)\s+(?P<age>\d+)) would create results like [{'name': 'Bob', 'age': 32}]
+
+    Do not give nested groups the same name as another values in the template.
```

Comment:
Can you either describe the behaviour, and/or handle it gracefully?

Also: s/another/other